tests/unit/test_io_hdf5_h5tools.py

```diff
@@ -164,6 +164,30 @@ def test_write_dataset_list(self):
         dset = self.f['test_dataset']
         self.assertTrue(np.all(dset[:] == a))
 
+    def test_write_dataset_lol_strings(self):
+        a = [['aa', 'bb'], ['cc', 'dd']]
+        self.io.write_dataset(self.f, DatasetBuilder('test_dataset', a, attributes={}))
+        dset = self.f['test_dataset']
+        decoded_dset = [[item.decode('utf-8') if isinstance(item, bytes) else item for item in sublist]
+                        for sublist in dset[:]]
+        self.assertTrue(decoded_dset == a)
+
+    def test_write_dataset_list_compound_datatype(self):
```

Comment:
There was missing code coverage for compound data types in the `__list_fill__` method in `src/hdmf/backends/hdf5/h5tools.py` so I added this test that I think should cover that case.

Reply:
Looks good!